server/code_generator.py

```diff
@@ -0,0 +1,34 @@
+import random
+import string
+
+bad_words_file = open('server/bad_words.txt', 'r')
+BAD_WORDS_LIST = bad_words_file.read().split('\n')
+bad_words_file.close()
+
+def generate_code(num_of_chars):
+    code = ''
+
+    while not is_valid(code):
+        code = generate_random_elems(num_of_chars)
+
+    return code
+
+def generate_random_elems(num_of_chars):
+    POSSIBLE_CHARS = string.ascii_letters + string.digits
+    code = ''
+
+    for x in range(num_of_chars):
+        random_elem = random.choice(POSSIBLE_CHARS)
+        code += random_elem
+
+    return code
+
+def is_valid(code):
+    if not code:
```

Comment:
Similarly to how we'll never pass in an empty string or a string with spaces, it's fair to assume we'll never pass a None value into this function. I think this check can be removed

Reply:
this is a situation where type annotations would help document valid inputs :smiley: